### PR TITLE
[202205] Update xcvrd to use new STATE_DB FAST_REBOOT entry

### PR DESCRIPTION
**Pull request by arfeigin**
This PR is similar to https://github.com/sonic-net/sonic-platform-daemons/pull/335, dedicated to 202205 branch.

Update xcvrd to check if fast-reboot is enabled according to the new value for FAST_REBOOT entry in STATE_DB.

This PR should come along with the following PRs:
https://github.com/sonic-net/sonic-buildimage/pull/14143
https://github.com/sonic-net/sonic-sairedis/pull/1217
https://github.com/sonic-net/sonic-utilities/pull/2724

This set of PRs solves the issue https://github.com/sonic-net/sonic-buildimage/issues/13251

Description
Update xcvrd to check the updated form of fast-reboot entry in state-db as it was changed.

Motivation and Context
Introducing fast-reboot finalizer on top of warmboot-finalizer, fast-reboot entry in STATE_DB is now changed from "1"/None to "enable: true/false".

How Has This Been Tested?
Existing tests, and fast-reboot.

Additional Information (Optional)

### CODE REVIEW

**Review by vaibhavhd — APPROVED** [review]

File: sonic-xcvrd/xcvrd/xcvrd.py [hunk 1]
```diff
@@ -923,12 +923,12 @@ def init_port_sfp_status_tbl(port_mapping, xcvr_table_helper, stop_event=threadi
...
     fastboot_enabled = False
     state_db_host =  daemon_base.db_connect("STATE_DB")
-    fastboot_tbl = swsscommon.Table(state_db_host, 'FAST_REBOOT')
+    fastboot_tbl = swsscommon.Table(state_db_host, 'FAST_RESTART_ENABLE_TABLE')
     keys = fastboot_tbl.getKeys()
 
     if "system" in keys:
-        output = subprocess.check_output('sonic-db-cli STATE_DB get "FAST_REBOOT|system"', shell=True, universal_newlines=True)
-        if "1" in output:
+        output = subprocess.check_output(['sonic-db-cli', 'STATE_DB', 'hget', "FAST_RESTART_ENABLE_TABLE|system", 'enable'], universal_newlines=True)
```
Comment (vaibhavhd) [inline sonic-xcvrd/xcvrd/xcvrd.py:930]: This PR is removing ` shell=True,` option. Please verify if that's intentional or merely due to copying a similar logic from master where ` shell=True,` is not used originally.

Comment (arfeigin) [reply sonic-xcvrd/xcvrd/xcvrd.py]: Modified, thanks.

**Comment by prgeor** [comment]
@arfeigin is cherry pick failing because of which this PR needs to be raised separately for 202205?

**Comment by arfeigin** [comment]
> @arfeigin is cherry pick failing because of which this PR needs to be raised separately for 202205?

In this repo not, we raised dedicated set of PRs for 202205 branch as we needed for a new updated 202205 version.

**Comment by stepanblyschak** [comment]
@vaibhavhd Please note, there is a submodule update to pick this change in 202205 - https://github.com/sonic-net/sonic-buildimage/pull/14275

**Pull request merged**